core/src/test/java/kafka/security/JaasTestUtils.java

```diff
@@ -0,0 +1,482 @@
+
+        return Arrays.asList(zkServerSection, zkClientSection);
+    }
+
+    public static JaasSection kafkaServerSection(String contextName, List<String> mechanisms, Optional<File> keytabLocation) {
+        List<JaasModule> modules = new ArrayList<>();
+        for (String mechanism : mechanisms) {
+            switch (mechanism) {
+                case "GSSAPI":
+                    modules.add(new Krb5LoginModule(
+                            true,
+                            true,
+                            keytabLocation.orElseThrow(() -> new IllegalArgumentException("Keytab location not specified for GSSAPI")).getAbsolutePath(),
+                            KAFKA_SERVER_PRINCIPAL,
+                            true,
+                            Optional.of(SERVICE_NAME),
+                            IS_IBM_SECURITY
+                    ));
+                    break;
+                case "PLAIN":
+                    Map<String, String> validUsers = new HashMap<>();
+                    validUsers.put(KAFKA_PLAIN_ADMIN, KAFKA_PLAIN_ADMIN_PASSWORD);
+                    validUsers.put(KAFKA_PLAIN_USER, KAFKA_PLAIN_PASSWORD);
+                    validUsers.put(KAFKA_PLAIN_USER_2, KAFKA_PLAIN_PASSWORD_2);
+                    modules.add(new PlainLoginModule(KAFKA_PLAIN_ADMIN, KAFKA_PLAIN_ADMIN_PASSWORD, false, validUsers));
+                    break;
+                case "OAUTHBEARER":
+                    modules.add(new OAuthBearerLoginModule(KAFKA_OAUTH_BEARER_ADMIN, false));
+                    break;
+                default:
+                    if (ScramMechanism.fromMechanismName(mechanism) != ScramMechanism.UNKNOWN) {
+                        modules.add(new ScramLoginModule(KAFKA_SCRAM_ADMIN, KAFKA_SCRAM_ADMIN_PASSWORD, false, new HashMap<>()));
+                    } else {
+                        throw new IllegalArgumentException("Unsupported server mechanism " + mechanism);
+                    }
+                    break;
+            }
+        }
+        return new JaasSection(contextName, modules);
+    }
+
+    private static JaasModule kafkaClientModule(String mechanism, Optional<File> keytabLocation, String clientPrincipal, String plainUser, String plainPassword, String scramUser, String scramPassword, String oauthBearerUser, String serviceName) {
+        switch (mechanism) {
+            case "GSSAPI":
+                return new Krb5LoginModule(
+                        true,
+                        true,
+                        keytabLocation.orElseThrow(() -> new IllegalArgumentException("Keytab location not specified for GSSAPI")).getAbsolutePath(),
+                        clientPrincipal,
+                        true,
+                        Optional.of(serviceName),
+                        IS_IBM_SECURITY
+                );
+            case "PLAIN":
```

Comment:
Could you please use `PlainSaslServer.PLAIN_MECHANISM` instead? and other constants should be replaced by existent variable too.